### DISCUSSION TITLE
sprite.fillRectVGradient not working

### DISCUSSION THREAD

**Discussion by GithubRick66** [Bodmer/TFT_eSPI#2275] [Q&A - Sprites]
Hello.
First, thank you to Bodmer for the fantastic library !

I am trying to use fillRectVGradient in a sprite and it don't seem to work
I am using the LilyGO TTGO T-Display-S3 ESP32-S3 board
tft.fillRectHGradient(0, 60, 160, 50, TFT_WHITE, TFT_BLACK);  is working
tft.fillRectVGradient(0, 60, 160, 50, TFT_WHITE, TFT_BLACK);  is working
sprite.fillRectHGradient(0, 60, 160, 50, TFT_WHITE, TFT_BLACK);  is working
sprite.fillRectVGradient(0, 60, 160, 50, TFT_WHITE, TFT_BLACK);  is not working

Excuse me if I have missed something

**Comment by Bodmer** [comment]
I woks fr me. Maybe you are using an old version of the library? Try v2.4.79

**Reply by GithubRick66** [reply]
Thanks for your reaction !

I am using the TFT_eSPI version supplied with the recommended library for the LilyGO TTGO T-Display-S3 ESP32-S3 board
found at https://github.com/Xinyuan-LilyGO/T-Display-S3/tree/main/lib/TFT_eSPI
I tried to replace this version of TFT_eSPI with the latest version from your Github page and adjusted the settings
I can't get it to work, the display remains black, I am not doing anything with IO jet, just the display
did you use a LilyGO TTGO T-Display-S3 ESP32-S3 board or another board to test or should that make no difference

Thanking you in advance for your effort

**Reply by Pokeystuff** [reply]
See the file here https://github.com/Xinyuan-LilyGO/TTGO-T-Display/blob/master/TFT_eSPI/User_Setup_Select.h 
and ensure these lines are correct
Around line 22: comment out as follows: // #include <User_Setup.h> 
Around line 53: #include <User_Setups/Setup25_TTGO_T_Display.h>
